internal/mediadb/media.go

```diff
@@ -0,0 +1,73 @@
+package mediadb
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+)
+
+type fileInfo struct {
+	key      string
+	mimeType string
+}
+
+func (m *mediaDB) GetAssetLocation(ctx context.Context, mediaID, mimeType string) (string, error) {
+	fileInfo, err := m.getFileInfo(mediaID, mimeType)
+	if err != nil {
+		logrus.WithError(err).Error("failed to get file info for cloudfront")
+		return "", err
+	}
+
+	return strings.Join([]string{m.s3BucketConfig.CloudFrontURL, fileInfo.key}, "/"), nil
+}
+
+func (m *mediaDB) UploadMedia(ctx context.Context, filename string, media []byte) (string, error) {
+	mimeType := http.DetectContentType(media[:512])
```

Comment:
It would be weird but we should be sure that the media has at least 512 bytes. Someone could upload an empty gif (0x0) that's like 128 bytes or less I think.

Reply:
should we return an err since we cannot detect type?